packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -240,6 +240,11 @@ export class RunningSummarizer implements IDisposable {
         await this.summarizingLock;
     }
 
+    public async stopOnDemand() {
+        await this.waitStop(false);
```

Comment:
waitStop() is weird as it returns promise (that waits for current summary to stop) only when it is called first time but returns right away if called again. I do not remember if I made it that way or if it was that way all the time. Note that while it's public, it's not on ISummarizer, so not really exposed to consumers.

I think based on what I see, I'd be surprised at these behaviors as a user when using newly added stopOnDemand() API. I'd expect some consistency here.

Reply:
Resolved this by accessing waitStop privately, on resolution of the summarizer's stopDeferred promise.